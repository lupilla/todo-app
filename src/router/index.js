import { createRouter, createWebHistory } from "vue-router";

/**
 * Array of routes included in the application
 * - Default route ("/"): all todos are showed
 * - Active route ("/active"): only active todos are showed
 * - Completed route ("/completed"): only completed todos are showed
 * - Page not found route: error page 404 is showed when the route does not match
 */
const routes = [
  {
    path: "/",
    name: "allTodos",
    component: () =>
      import(
        /* webpackChunkName: 'alltodos'*/ "@/modules/todos/pages/TodoPage.vue"
      ),
  },
  {
    path: "/active",
    name: "activeTodos",
    component: () =>
      import(
        /* webpackChunkName: 'active'*/ "@/modules/todos/pages/TodoPage.vue"
      ),
  },
  {
    path: "/completed",
    name: "completedTodos",
    component: () =>
      import(
        /* webpackChunkName: 'completed'*/ "@/modules/todos/pages/TodoPage.vue"
      ),
  },
  {
    path: "/:pathMatch(.*)*",
    component: () =>
      import(
        /* webpackChunkName: 'NoPageFoundPage'*/ "@/modules/shared/pages/NoPageFound.vue"
      ),
  },
];

/**
 * Router creation method to export it to the main.js file
 */
const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
