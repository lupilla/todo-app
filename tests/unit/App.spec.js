import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";
import createVuexStore from "./mocks/mock-store.js";
import router from "@/router";
const store = createVuexStore();
const $route = {
  path: "/",
};
describe("App tests", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallowMount(App, {
      global: {
        mocks: {
          $route,
        },
        plugins: [store, router],
      },
    });
  });
  test("should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should have a div with an id with 'app' value", () => {
    const id = wrapper.find("#app");
    expect(id.exists()).toBe(true);
    expect(id.isVisible()).toBe(true);
  });

  test("should have a router-view-stub with name 'default'", () => {
    const routerStub = wrapper.find("router-view-stub");
    expect(routerStub.exists()).toBe(true);
    expect(routerStub.isVisible()).toBe(true);
  });
});
