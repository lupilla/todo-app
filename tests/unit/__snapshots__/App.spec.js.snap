// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match with snapshot 1`] = `
<div id="app" class="todoapp">
  <router-view-stub name="default"></router-view-stub>
</div>
`;
